kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -103,7 +116,16 @@ public Class realClass(String elementName) {
                 };
             }
         });
+    }
 
+    private MapperWrapper chainMapperWrappers(List<XStreamMarshallerExtension> extensions, MapperWrapper last) {
+        if (extensions.isEmpty()) {
+            return last;
+        } else {
+            XStreamMarshallerExtension head = extensions.get(0);
+            List<XStreamMarshallerExtension> tail = extensions.subList(1, extensions.size());
```

Comment:
Yeah, it may be a bit easier to read. I was trying not to mutate the provided params as I consider that a bad practice. I can change this though.